src/test/java/org/ambraproject/wombat/service/PeerReviewServiceImplTest.java

```diff
@@ -89,7 +89,48 @@ public void testAsHtml() throws IOException {
         )).build();
 
 
-    PeerReviewService serviceWithMockedContent = new PeerReviewServiceImpl() {
+    PeerReviewService serviceWithMockedContent = getServiceWithMockedContent();
+
+    String html = serviceWithMockedContent.asHtml(itemTable);
+    Document d = Jsoup.parse(html);
+
+    assertThat(d.select(".review-history .revision").get(0).text(), containsString("Original Submission"));
+    assertThat(d.select(".review-history .revision").get(0).text(), containsString("June 1, 2018"));
+    assertThat(d.select(".review-history .decision-letter").get(0).text(), containsString("InitialDecisionLetterSampleBody"));
+    assertThat(d.select(".review-history .revision").get(1).text(), containsString("Revision 1"));
+    assertThat(d.select(".review-history .author-response").get(0).text(), containsString("FirstRoundAuthorResponseSampleBody"));
+    assertThat(d.select(".review-history .decision-letter").get(1).text(), containsString("FirstRoundDecisionLetterSampleBody"));
+    assertThat(d.select(".review-history .revision").get(2).text(), containsString("Revision 2"));
+    assertThat(d.select(".review-history .author-response").get(1).text(), containsString("SecondRoundAuthorResponseSampleBody"));
+    assertThat(d.select(".review-history .decision-letter").get(2).text(), containsString("SecondRoundDecisionLetterSampleBody"));
+    assertThat(d.select(".review-history .revision").get(3).text(), containsString("Formally Accepted"));
+    assertThat(d.select(".review-history .acceptance-letter").get(0).text(), containsString("AcceptanceLetterSampleBody"));
+  }
+
+  @Test
+  public void testSubarticleDoi() throws IOException {
+    ImmutableMap<String, ? extends Map<String, ?>> itemTable = new ImmutableMap.Builder<String, Map<String, ?>>()
+        .put("10.1371/journal.pone.0207232.r001", ImmutableMap.of(
+            "doi", "10.1371/journal.pone.0207232.r001",
+            "itemType", "reviewLetter",
+            "files", ImmutableMap.of("letter", ImmutableMap.of(
+                "crepoKey", "info:doi/10.1371/journal.pone.0207232.r001.xml",
+                "crepoUuid", UUID.randomUUID().toString()
+            ))
+        )).build();
+
+
+    PeerReviewService serviceWithMockedContent = getServiceWithMockedContent();
+
+    String html = serviceWithMockedContent.asHtml(itemTable);
+    System.out.println(html);
+    Document d = Jsoup.parse(html);
+
+    assertThat(d.select(".review-history .review__doi").get(0).text(), containsString("https://doi.org/10.1371/journal.pone.0207232.r001"));
+  }
+
+  PeerReviewServiceImpl getServiceWithMockedContent() {
+    return new PeerReviewServiceImpl() {
```

Comment:
It's kind of a bad pattern to stub/mock/spy on the class under test. Is there some way that we can avoid this?

Reply:
Alternatives might not come cheap. I'll timebox an attempt.

Reply:
The primary [method being mocked](https://github.com/PLOS/wombat/blob/36bb47eb2755d40d21aad46b102e8dad1a24ff98/src/main/java/org/ambraproject/wombat/service/PeerReviewServiceImpl.java#L243) is intended to encapsulate http connection and streaming methods.  So, it's existence already a change to accommodate tests, which is sometimes a smell but I think appropriate in this case.  I could extract a utility class to get content from the contentApi as a string so that the mocked collaborator is a class instead of a method.  

Mocking the individual http and streaming collaborators would be more challenging and I'm not sure about the benefit.

Reply:
OK, I see your point. :) Now is probably not the time to rearchitect this, but the impossibility of mocking out everything but the class under test here makes me wonder about some of the architecture underlying this. I started trying to do it, but I see your point - there is not a clear separation between the underlying HTTP code and the data model used here (`ContentKey`, etc) which is making this much more complicated that necessary. Oh well. :(

Reply:
Well, now you've left me eager to practice the sorts of improvements you made to these tests.  :)  

It seems like a lot more test boilerplate if I were to extract a `ContentApiUtil` class, or alternatively move the logic to a `corpusContentApi.getAsString()`.  Does this sound like the correct approach to you if I needed to mock the content of a collaborating class? https://stackoverflow.com/questions/35086806/use-different-spring-test-context-configuration-for-different-test-methods.  Might be easier to just create a different test class per context.

Reply:
Well, feel free to give it a shot :) I tried, but I found that the `CorpusContentApi` has an unfortunate API that mixes low-level HTTP with high-level stuff like `ContentKey`. This makes it hard because you can't just mock out the HTTP stuff or the high level stuff, but both mixed together, and you have to set up the relationships between the two mocked bits properly.

(This is what I like most about tests, they really help you figure out the awkward parts of the API, because if you find yourself mocking out this much, it seems to indicate to me an awkward underlying API).